### PR TITLE
Source Pinterest - Adding missing columns for analytics streams

### LINKED ISSUE #22206: Source Pinterest: Extend metrics in analytics stream
## Environment
OS Version / Instance: Ubuntu
Deployment: Docker
Source Connector and version: source-pinterest:0.1.9

## Current Behavior
We found below metrics missing in the connector:

- VIDEO_P0_COMBINED_1
- VIDEO_P100_COMPLETE_1
- VIDEO_P25_COMBINED_1
- VIDEO_P50_COMBINED_1
- VIDEO_P75_COMBINED_1
- VIDEO_P95_COMBINED_1
- VIDEO_3SEC_VIEWS_1
- VIDEO_MRC_VIEWS_1
- APP_INSTALL_COST_PER_ACTION
- CLICKTHROUGH_2
- CLICKTHROUGH_1
- ENGAGEMENT_2
- ENGAGEMENT_1
- TOTAL_CLICK_ADD_TO_CART

## Are you willing to submit a PR?
Yes

### PR DESCRIPTION
**Pull request by artusiep**
## What
*Adding missing metrics for pinterest source*

Not all columns are available for synchronous endpoint so I have added all available columns:
```
- ADVERTISER_ID                   |  #21326
- CLICKTHROUGH_1                  |  #22206 and #21326
- CLICKTHROUGH_1_GROSS            |  #21326
- CLICKTHROUGH_2                  |  #22206 and #21326
- ENGAGEMENT_1                    |  #22206 and #21326
- ENGAGEMENT_2                    |  #22206 and #21326
- PIN_ID                          |  #21326
- PIN_PROMOTION_ID                |  #21326
- REPIN_1                         |  #21326
- REPIN_2                         |  #21326
- TOTAL_CLICK_ADD_TO_CART         |  #22206 and #21326
- TOTAL_CLICK_LEAD                |  #21326
- TOTAL_CUSTOM                    |  #21326
- TOTAL_ENGAGEMENT                |  #21326
- TOTAL_ENGAGEMENT_LEAD           |  #21326
- TOTAL_LEAD                      |  #21326
- TOTAL_OFFLINE_CHECKOUT          |  #21326
- TOTAL_VIEW_ADD_TO_CART          |  #21326
- TOTAL_VIEW_LEAD                 |  #21326
- TOTAL_WEB_SESSIONS              |  #21326
- VIDEO_LENGTH                    |  #21326
- WEB_SESSIONS_1                  |  #21326
- WEB_SESSIONS_2                  |  #21326
- VIDEO_P0_COMBINED_1             |  Not Available in Synchronous API call
- VIDEO_P100_COMPLETE_1           |  Not Available in Synchronous API call
- VIDEO_P25_COMBINED_1            |  Not Available in Synchronous API call
- VIDEO_P50_COMBINED_1            |  Not Available in Synchronous API call
- VIDEO_P75_COMBINED_1            |  Not Available in Synchronous API call
- VIDEO_P95_COMBINED_1            |  Not Available in Synchronous API call
- VIDEO_3SEC_VIEWS_1              |  Not Available in Synchronous API call
- VIDEO_MRC_VIEWS_1               |  Not Available in Synchronous API call
- APP_INSTALL_COST_PER_ACTION     |  Not Available in Synchronous API call
```
This contribution brings 22 columns to analytics columns.

`windows_in_days` parameter is not used anywhere, and it is not configurable so deleting that from conftest.

Unit test:
image

Closes  #22206 #21326
Duplicates: [https://github.com/airbytehq/airbyte/pull/22501]

## How
Adding columns and making list of columns easier to review

## Recommended reading order
1. `airbyte-integrations/connectors/source-pinterest/source_pinterest/utils.py`
2. `airbyte-integrations/connectors/source-pinterest/source_pinterest/source.py`

## 🚨 User Impact 🚨

Pinterest source users would need to `refresh source schema` to pickup changes

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

### CODE REVIEW

**Comment by artusiep** [comment]
Tagging @sh4sh, @natalyjazzviolin

**Comment by artusiep** [comment]
Acceptance tests:
image

I do not really know how to make this change backward compatibile

**Comment by artusiep** [comment]
Tested with Airbyte docker 
image

**Comment by artusiep** [comment]
@sh4sh, @natalyjazzviolin

**Review by sajarin — COMMENTED** [review]
Hey @artusiep, 

Thanks for the contribution, it looks like you're adding around 22 new columns to the analytics streams. 

In the conftest.py, what's the reason behind removing the "windows_in_days" value? 

Also there seems to be some merge conflicts, can you fix them?

**Comment by artusiep** [comment]
Hey @sajarin 

answering your question

> Hey @artusiep,
> 
> Thanks for the contribution, it looks like you're adding around 22 new columns to the analytics streams.

Exactly this is the main purpose of this contribution

> In the conftest.py, what's the reason behind removing the "windows_in_days" value?

`windows_in_days` parameter is simply not used anywhere, and it is not configurable. Take a look at: https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-pinterest/source_pinterest/source.py#L44-L46 I just wanted do make a clean up of this variable as well 

> Also there seems to be some merge conflicts, can you fix them?

Merge conflict fixed

**Comment by artusiep** [comment]
@sajarin I think the backward compatibility error is connected with adding these columns and changing the type of columns. 
Changed type because it was failing on dbt normalization in BigQuery:
https://github.com/airbytehq/airbyte/pull/23457/files#diff-65c6cd00462455b47cc1ddbe4c50322833e2d6cb2c6c4997012d1c9d2c8f3b12L33-L35
https://github.com/airbytehq/airbyte/pull/23457/files#diff-254365ad0710b775423057cada068e509205a6a37317b9da247e7f6246893977L15-L20
https://github.com/airbytehq/airbyte/pull/23457/files#diff-a8cf2c89813a5bbf6f0ba32e0c2107c7630f38fbd7d86e53159f235b2423ceadL15-L17

**Comment by artusiep** [comment]
Could @airbytehq/community-assistance-dri take a look?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-pinterest/integration_tests/sample_config.json [hunk 1]
```diff
@@ -2,6 +2,5 @@
   "client_id": "client_id",
   "client_secret": "client_secret",
   "refresh_token": "refresh_token",
-  "start_date": "2020-06-01",
-  "window_in_days": 1
```
[the remark this answers is not in the document]
Comment (artusiep) [reply airbyte-integrations/connectors/source-pinterest/integration_tests/sample_config.json:6]: As it is mentioned above: 

> In the conftest.py, what's the reason behind removing the "windows_in_days" value?

`windows_in_days` parameter is simply not used anywhere, and it is not configurable. Take a look at: https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-pinterest/source_pinterest/source.py#L44-L46 I just wanted do make a clean up of this variable as well

I can rollback that but it is not going to be used

File: airbyte-integrations/connectors/source-pinterest/source_pinterest/schemas/ad_account_analytics.json [hunk 2]
```diff
@@ -31,11 +34,20 @@
       "type": ["null", "number"]
     },
     "CAMPAIGN_NAME": {
-      "type": ["null", "number"]
+      "type": ["null", "string"]
```
[the remark this answers is not in the document]
Comment (artusiep) [reply airbyte-integrations/connectors/source-pinterest/source_pinterest/schemas/ad_account_analytics.json:37]: I will try to understand how this backward compatibility acceptance tests works in order to mitigate that

[the remark this answers is not in the document]
Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-pinterest/source_pinterest/schemas/ad_account_analytics.json:37]: Hello @artusiep sorry the long delay here. I submitted a few changes to correct integration tests.

**Review by marcosmarxm — APPROVED** [review]
Thanks @artusiep I'll request the final approval from the connector team to release the new version of your changes. This is going to happens next week but everything is ok from your side. Sorry again for the long delay to review the contribution.

**Review by lazebnyi — APPROVED** [review]
LGTM! Need update strictness level for tests

File: airbyte-integrations/connectors/source-pinterest/acceptance-test-config.yml [hunk 3]
```diff
@@ -62,4 +62,4 @@ acceptance_tests:
           disable_for_version: 0.1.2
         spec_path: source_pinterest/spec.json
 connector_image: airbyte/source-pinterest:dev
-test_strictness_level: high
+test_strictness_level: low
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-pinterest/acceptance-test-config.yml:65]: `strictness_level: high` -  required for GA connectors

**Review by lazebnyi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-pinterest/acceptance-test-config.yml (hunk 3 above)
Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-pinterest/acceptance-test-config.yml:65]: Sorry I lower the level to test only test_read. Rolling back right now.

**Review by lazebnyi — APPROVED** [review]
LGTM!

**Pull request merged**